### ISSUE TITLE
[macOS] Crash when starting CUO with login music enabled

### ISSUE THREAD

**Issue by canterberry** [ClassicUO/ClassicUO#600]
> 💡 Duplicates #483.

**Describe the bug**

When ClassicUO attempts to play a sound for the first time (whether at the login screen or upon entering the world after character selection), it will crash on Mac OS X if it cannot find an audio device.

A separate, but related, issue, is that ClassicUO cannot find any audio devices on my Mac, even though no other applications have difficulty with this (e.g: OBS Studio, VLC, Spotify).

**To Reproduce**

1. Launch ClassicUO on a Mac with `"login_music": true` in *settings.json*.
2. Client crashes immediately with the following exception:

```
9:13:15 PM |  Panic   | Microsoft.Xna.Framework.Audio.NoAudioHardwareException (0x80004005): External component has thrown an exception.
  at Microsoft.Xna.Framework.Audio.SoundEffect.Device ()[0x0001e] in <fc394c8506654ff0a1d351c865b1e88b>:0 
  at Microsoft.Xna.Framework.Audio.SoundEffectInstance..ctor (Microsoft.Xna.Framework.Audio.SoundEffect parent)[0x00018] in <fc394c8506654ff0a1d351c865b1e88b>:0 
  at Microsoft.Xna.Framework.Audio.DynamicSoundEffectInstance..ctor (System.Int32 sampleRate, Microsoft.Xna.Framework.Audio.AudioChannels channels)[0x00000] in <fc394c8506654ff0a1d351c865b1e88b>:0 
  at ClassicUO.IO.Audio.Sound.GetNewInstance (System.Boolean asEffect)[0x00018] in /path/to/ClassicUO/src/IO/Audio/Sound.cs:202 
  at ClassicUO.IO.Audio.Sound.Play (System.Boolean asEffect, ClassicUO.IO.Audio.AudioEffects effect, System.Single volume, System.Boolean spamCheck)[0x00036] in /path/to/ClassicUO/src/IO/Audio/Sound.cs:133 
  at ClassicUO.Game.Managers.AudioManager.PlayMusic (System.Int32 music)[0x000c4] in /path/to/ClassicUO/src/Game/Managers/AudioManager.cs:102 
  at ClassicUO.Network.PacketHandlers.Season (ClassicUO.Network.Packet p)[0x00027] in /path/to/ClassicUO/src/Network/PacketHandlers.cs:2710 
  at ClassicUO.Network.PacketHandlers.OnPacket (System.Object sender, ClassicUO.Network.Packet p)[0x0001b] in /path/to/ClassicUO/src/Network/PacketHandlers.cs:95 
  at ClassicUO.Utility.Exstentions.Raise[T] (System.EventHandler`1[TEventArgs] handler, T e, System.Object sender) [0x00000] in /path/to/ClassicUO/src/Utility/Exstentions.cs:44 
  at ClassicUO.Network.NetClient.Update() [0x00066] in /path/to/ClassicUO/src/Network/NetClient.cs:250 
  at ClassicUO.Engine.OnNetworkUpdate(System.Double totalMS, System.Double frameMS) [0x00052] in /path/to/ClassicUO/src/Engine.cs:818 
  at ClassicUO.Engine.Tick() [0x00034] in /path/to/ClassicUO/src/Engine.cs:689 
  at Microsoft.Xna.Framework.SDL2_FNAPlatform.RunLoop(Microsoft.Xna.Framework.Game game) [0x006a9] in <fc394c8506654ff0a1d351c865b1e88b>:0 
  at Microsoft.Xna.Framework.Game.Run() [0x0002c] in <fc394c8506654ff0a1d351c865b1e88b>:0 
  at ClassicUO.Engine.Run(System.String[] args) [0x00019] in /path/to/ClassicUO/src/Engine.cs:319 
  at ClassicUO.Bootstrap.Main(System.String[] args) [0x0000e] in /path/to/ClassicUO/src/Main.cs:25 
Thread started: <Thread Pool> #12
exception inside UnhandledException handler: (null) assembly:/Library/Frameworks/Mono.framework/Versions/5.18.1/lib/mono/4.5/mscorlib.dll type:InvalidOperationException member:(null)

9:13:15 PM |  Trace   | Saving path:		/path/to/ClassicUO/bin/Release/Data/Profiles/...
9:13:15 PM |  Trace   | Saving done!
9:13:15 PM |  Warning | PlayerMobile disposed!
[ERROR] FATAL UNHANDLED EXCEPTION: Microsoft.Xna.Framework.Audio.NoAudioHardwareException(0x80004005): External component has thrown an exception.
 at Microsoft.Xna.Framework.Audio.SoundEffect.Device ()[0x0001e] in <fc394c8506654ff0a1d351c865b1e88b>:0 

 at Microsoft.Xna.Framework.Audio.SoundEffectInstance..ctor (Microsoft.Xna.Framework.Audio.SoundEffect parent)[0x00018] in <fc394c8506654ff0a1d351c865b1e88b>:0 

 at Microsoft.Xna.Framework.Audio.DynamicSoundEffectInstance..ctor (System.Int32 sampleRate, Microsoft.Xna.Framework.Audio.AudioChannels channels)[0x00000] in <fc394c8506654ff0a1d351c865b1e88b>:0 

 at ClassicUO.IO.Audio.Sound.GetNewInstance (System.Boolean asEffect)[0x00018] in /path/to/ClassicUO/src/IO/Audio/Sound.cs:202 

 at ClassicUO.IO.Audio.Sound.Play (System.Boolean asEffect, ClassicUO.IO.Audio.AudioEffects effect, System.Single volume, System.Boolean spamCheck)[0x00036] in /path/to/ClassicUO/src/IO/Audio/Sound.cs:133 

 at ClassicUO.Game.Managers.AudioManager.PlayMusic (System.Int32 music)[0x000c4] in /path/to/ClassicUO/src/Game/Managers/AudioManager.cs:102 

 at ClassicUO.Network.PacketHandlers.Season (ClassicUO.Network.Packet p)[0x00027] in /path/to/ClassicUO/src/Network/PacketHandlers.cs:2710 

 at ClassicUO.Network.PacketHandlers.OnPacket (System.Object sender, ClassicUO.Network.Packet p)[0x0001b] in /path/to/ClassicUO/src/Network/PacketHandlers.cs:95 

 at ClassicUO.Utility.Exstentions.Raise[T] (System.EventHandler`1[TEventArgs] handler, T e, System.Object sender) [0x00000] in /path/to/ClassicUO/src/Utility/Exstentions.cs:44 
  at ClassicUO.Network.NetClient.Update() [0x00066] in /path/to/ClassicUO/src/Network/NetClient.cs:250 
  at ClassicUO.Engine.OnNetworkUpdate(System.Double totalMS, System.Double frameMS) [0x00052] in /path/to/ClassicUO/src/Engine.cs:818 
  at ClassicUO.Engine.Tick() [0x00034] in /path/to/ClassicUO/src/Engine.cs:689 
  at Microsoft.Xna.Framework.SDL2_FNAPlatform.RunLoop(Microsoft.Xna.Framework.Game game) [0x006a9] in <fc394c8506654ff0a1d351c865b1e88b>:0 
  at Microsoft.Xna.Framework.Game.Run() [0x0002c] in <fc394c8506654ff0a1d351c865b1e88b>:0 
  at ClassicUO.Engine.Run(System.String[] args) [0x00019] in /path/to/ClassicUO/src/Engine.cs:319 
  at ClassicUO.Bootstrap.Main(System.String[] args) [0x0000e] in /path/to/ClassicUO/src/Main.cs:25 
```

**Expected behavior**

Two things:
 1. I have audio devices on my Mac that other apps are able to discover with no difficulty. ClassicUO should be able to discover them.
 2. If, for some reason, ClassicUO cannot discover an audio device, it should not panic and crash. Instead, it should log a warning and refrain from further attempts to play audio.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: OS X 10.14.5
 - ClassicUO Version: Latest `master` from source *and* 0.0.1.11, the latest prebuilt release as of this writing.

**Comment by canterberry** [comment]
Discovered the cause of the problem. I needed to install SDL2 (`brew install sdl2`).
After trying to work around the issue by disabling sound and music by modifying the profile settings as well as the top-level settings, I discovered that the Options gump would not load, either, due to an inability to read a PNG and a more informative error message identifying SDL2 as missing.

I'll go ahead and close my own issue now.

Maybe it's mentioned in the README already that SDL2 is required on Mac OS X. If not, it probably should be. Either way, hopefully if anyone else runs into this, this may help them!

❤️
